### DISCUSSION TITLE
How to use qsort in C file ; ERROR at link time

### DISCUSSION THREAD

**Discussion by benoitLecuppe** [micropython/micropython#10574] [General]
Hello,
I'm currently customising "/micropython/ports/esp8266/esppwm.c" and i want to use qsort() from <stdlib.h>.

I have an error during link time :

```
build-GENERIC/esppwm.o: In function `pwm_dump':
esppwm.c:(.irom0.text+0x2b4): undefined reference to `qsort'
build-GENERIC/esppwm.o: In function `timetable_init':
/micropython/ports/esp8266/esppwm.c:253: undefined reference to `qsort'
make: *** [Makefile:214: build-GENERIC/firmware.elf] Error 1
```
I think I must add something into makefile to make it work but which one and how ?

Thanks for the help
Benoit (from France)

**Comment by jimmo** [comment]
> and i want to use qsort() from <stdlib.h>.

We do not link against a libc on most platforms. See `-nostdlib` in the makefile.

This prevents accidentally increasing code size by using additional functions.

MicroPython provides an implementation of quicksort for sorting lists -- `mp_obj_list_sort`. You can see this being used for example in `mp_help_print_modules` to print the sorted list of built-in modules.

**Comment by benoitLecuppe** [comment]
Ok thank you, i will look -nostdlib.
